diff --git a/src/runtime_src/core/common/debug.cpp b/src/runtime_src/core/common/debug.cpp
index a16b354a4..7632d9383 100644
--- a/src/runtime_src/core/common/debug.cpp
+++ b/src/runtime_src/core/common/debug.cpp
@@ -32,7 +32,6 @@ debugf(const char* format,...)
   debug_lock lk;
   va_list args;
   va_start(args,format);
-  printf("%lu: ",time_ns());
   vprintf(format,args);
   va_end(args);
 }
diff --git a/src/runtime_src/core/common/module_loader.cpp b/src/runtime_src/core/common/module_loader.cpp
index 46a28036c..beb295117 100644
--- a/src/runtime_src/core/common/module_loader.cpp
+++ b/src/runtime_src/core/common/module_loader.cpp
@@ -186,7 +186,7 @@ shim_path()
 #ifdef _WIN32
   path /= name + ".dll";
 #else
-  path /= "lib/lib" + name + ".so." + XRT_VERSION_MAJOR;
+  path /= "lib" + name + ".so." + XRT_VERSION_MAJOR;
 #endif
 
   if (!sfs::exists(path) || !sfs::is_regular_file(path))
